contracts/PricerRule.sol

```diff
@@ -0,0 +1,408 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import "./PriceOracleInterface.sol";
+import "./EIP20TokenInterface.sol";
+import "./TokenRules.sol";
+import "./SafeMath.sol";
+
+
+contract PricerRule is Organized {
+
+    /* Usings */
+
+    using SafeMath for uint256;
+
+
+    /* Events */
+
+    event PriceOracleAdded(
+        address _priceOracle
+    );
+
+    event PriceOracleRemoved(
+        address _priceOracle
+    );
+
+    event AcceptanceMarginAdded(
+        bytes3 _quoteCurrencyCode,
+        uint256 acceptanceMargin
+    );
+
+    event AcceptanceMarginRemoved(
+        bytes3 _quoteCurrencyCode
+    );
+
+
+    /* Storage */
+
+    /**
+     * @dev The code of the base currency of the economy.
+     */
+    bytes3 public baseCurrencyCode;
+
+    /**
+     * @dev EIP20 token address of the economy.
+     */
+    EIP20TokenInterface public eip20Token;
+
+    /**
+     * @dev Conversion rate from the base currency to the economy token.
+     */
+    uint256 public conversionRateFromBaseCurrencyToToken;
+
+    /**
+     * @dev Conversion rate decimals from the base currency to the
+     *      economy token.
+     */
+    uint256 public conversionRateDecimalsFromBaseCurrencyToToken;
+
+    /**
+     * @dev Token rules address of the economy.
+     */
+    TokenRules private tokenRules;
+
+    /**
+     * @dev Mapping from pay-currency code to corresponding price oracle.
+     *      The mapped price oracle's base currency code should be the economy
+     *      base currency code (see baseCurrencyCode storage variable)
+     *      and the quote currency code should be the corresponding pay-currency
+     *      code.
+     */
+    mapping(bytes3 => PriceOracleInterface) private baseCurrencyPriceOracles;
+
+    /**
+     * @dev Mapping from pay-currency code to price difference accepatance
+     *      margin. During a pay operation an intended price for
+     *      pay-currency is presented, that is checked against current price of
+     *      pay-currenency wrt stored acceptance margin.
+     */
+    mapping(bytes3 => uint256) private baseCurrencyPriceAcceptanceMargins;
+
+
+    /* Special Functions */
+
+    /**
+     * @notice Constructs a new pricer object.
+     *
+     * @dev Function requires:
+     *          - The economy token address is not null.
+     *          - The base currency code is not empty.
+     *          - Conversion rate from the base currency to the token is not 0.
+     *          - Conversion rate's decimals from the base currency to the
+     *            token is not 0.
+     *          - The economy token rules address is not null.
+     *
+     * @param _organization Organization address.
+     * @param _eip20Token The economy token address.
+     * @param _baseCurrencyCode The economy base currency code.
```

Comment:
`baseCurrencyCode` is not set—I assume it should be, but please disregard if it should not be.

Reply:
I've already fixed it locally during testing as my test was failing :). Thx for catching!